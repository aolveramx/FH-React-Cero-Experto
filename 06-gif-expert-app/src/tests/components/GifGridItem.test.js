import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import GifGridItem from "../../components/GifGridItem"

describe('Tests in GifGridItem.js', () => {

  const url = 'http://localhost/img.jpg'
  const title = 'Title'
  const wrapper = shallow(<GifGridItem title={title} url={url}/>)

  test('shoud render <GifGridItem />', () => {

    expect(wrapper).toMatchSnapshot()

  });

  test('should have paragraph with title', () => {

    const p = wrapper.find('p')
    expect(p.text().trim()).toBe(title)

  });

  test('should have same img at url & props', () => {
    
    const img = wrapper.find('img')

    expect(img.prop('src')).toBe(url)
    expect(img.prop('alt')).toBe(title)

  });

  test('should have className="animate__fadeIn"', () => {

    const div = wrapper.find('div')

    expect(div.hasClass('animate__fadeIn')).toBe(true)

    // const className = div.prop('className')
    // expect(className.includes('animate__fadeIn')).toBe(true) 

  });

})
