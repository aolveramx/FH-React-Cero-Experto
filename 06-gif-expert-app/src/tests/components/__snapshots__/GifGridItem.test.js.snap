// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifGridItem.js shoud render <GifGridItem /> 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Title"
    src="http://localhost/img.jpg"
  />
  <p>
    Title
  </p>
</div>
`;
